engine/src/gfx/particle.h

```diff
@@ -50,10 +50,10 @@ struct ParticlePoint
  */
 class ParticleTrail
 {
-    std::vector< Vector, aligned_allocator<Vector> > particleVel;
-    std::vector< QVector, aligned_allocator<QVector> > particleLoc;
-    std::vector< GFXColor, aligned_allocator<GFXColor> > particleColor;
-    std::vector< float, aligned_allocator<float> > particleSize;
+    std::vector< Vector > particleVel;
+    std::vector< QVector > particleLoc;
+    std::vector< GFXColor > particleColor;
+    std::vector< float > particleSize;
```

Comment:
Can we continue to use the `aligned_allocator`s here as well?

Reply:
`float` isn't a valid type for using with `aligned_allocator`.

Reply:
@BenjamenMeyer Do you have a reference link for that? I find that surprising.

Reply:
I believe that [this](https://github.com/vegastrike/Vega-Strike-Engine-Source/blob/master/engine/src/gfx/aligned.h) is the aligned_allocator intended to be used here. Not the C++17 `aligned_alloc`, if that is what you were thinking of. (Though we might eventually switch to the latter, if we decide to start using C++17 features at some point.)

Reply:
> @BenjamenMeyer Do you have a reference link for that? I find that surprising.

Or what was the specific error message(s)?

Reply:
@stephengtuggy the compilers refused to even build it as the types did not have the correct support for that functionality.

Reply:
OK. I think I found some of the error messages you were getting. Stuff like "Cannot use _Tp_Alloc here because it does not name a type" ? Among other things?

I got some of the same errors when I tried to build the project on Ubuntu Mantic for the first time. It looks like we're going to have to deal with this issue more and more as we move to the newer Linux versions.

I'm wondering if it's time to take a step back; rethink what the code was originally trying to do, using compiler-specific constructs and such; and see if there's a better way to accomplish that same goal, using C++11 standard constructs like some of those mentioned [here](https://en.cppreference.com/w/cpp/memory/align).

Reply:
IMHO we need to stick to standard C++ and not do the compiler specific route. There's no need anymore. Yeah, when the project first started one needed to do so because so much was custom to compilers and what is now standardized across compilers wasn't.

More often than not the compiler does the same right thing without hints. So I think we'll be good if we just let it do its thing.

Reply:
Definitely, let's stick to standard C++ as much as possible. However, I'm not convinced that we can get by without the alignment hints in this case.

I believe my latest code fixes the compile errors we were getting. Hopefully, those changes can be incorporated into this PR, for a solution that we can all be happy with.